### PR TITLE
Change approval to be needed only once for transferring the NFT durin…

### PR DESCRIPTION
**Pull request by philipstanislaus**
…g issue/finance, borrow/supply, repay, redeem/withdraw.

Signed-off-by: Philip Stanislaus <6912756+philipstanislaus@users.noreply.github.com>

### CODE REVIEW

File: services/tinlake/actions.ts [hunk 1]
```diff
@@ -307,26 +310,31 @@ export async function borrow(tinlake: any, loan: Loan, amount: string) {
 }
 
 // repay full loan debt
-export async function repay(tinlake: any, loan: Loan) {
+export async function repay(tinlake: ITinlake, loan: Loan) {
```
Comment (manuelpolzhofer) [inline services/tinlake/actions.ts:313]: 👍

File: .env.production [hunk 2]
```diff
@@ -0,0 +1,7 @@
+NEXT_PUBLIC_ENV=PROD
```
Comment (manuelpolzhofer) [inline .env.production:1]: I think this file shouldn't be part of the repo.

Ca

Comment (manuelpolzhofer) [reply .env.production:1]: It would be better to inject it with netlify

Comment (philipstanislaus) [reply .env.production:1]: I would disagree – the contents of .env.development and .env.production are public knowledge anyways, and including them here significantly simplifies onboarding of new engineers and communication in our team. E. g. when the structure changed and one team member is on vacation, it's very hard to continue developing/testing the repository.

Comment (manuelpolzhofer) [reply .env.production:1]: I don't have a strong opinion about it, if it contains no secrets and it makes it easier for you to develop at the current stage. However, if we move to a paid version of infura our link would be searchable if it is stored public on github. 

Obviously you could look it up if you use the tinlake-ui in the browser but not searchable via web.

Comment (philipstanislaus) [reply .env.production:1]: Fair point, I'll add a comment to the .env file to make sure we don't expose it.

Comment (manuelpolzhofer) [reply .env.production:1]: If we use the paid infura version we could us a reverse proxy like nginx

Comment (philipstanislaus) [reply .env.production:1]: Absolutely 💯

**Pull request merged**